blockchain/blockchain.go

```diff
@@ -637,6 +637,12 @@ func (bc *BlockChain) HasBlockAndState(hash common.Hash, number uint64) bool {
 	return bc.HasState(block.Root())
 }
 
+// ShouldTryInserting returns the state whether the block should be inserted.
+// If node don't have a block or after it's current state, it should be insert.
```

Comment:
```suggestion
// If a node doesn't have the given block or the block number of given block is higher than the node's head block, it can try inserting the block.
```